### PR TITLE
Add robin-map-dev

### PR DESCRIPTION
**Pull request by nachovizzo**
Please add the following dependency to the rosdep database.

## Package name:

robin-map

## Package Upstream Source:

https://github.com/Tessil/robin-map

## Purpose of using this:

Dependency used in projects like [KISS-ICP](https://github.com/PRBonn/kiss-icp) and not present in the rosdep index

Distro packaging links:

## Links to Distribution Packages

- Debian: https://packages.debian.org/
  - https://packages.debian.org/bookworm/robin-map-dev
- Ubuntu: https://packages.ubuntu.com/
  - https://packages.ubuntu.com/focal/robin-map-dev
- Fedora: https://packages.fedoraproject.org/
  - https://packages.fedoraproject.org/pkgs/robin-map/robin-map-devel/
- Arch: https://www.archlinux.org/packages/
  - https://archlinux.org/packages/extra/x86_64/robin-map/
- Gentoo: https://packages.gentoo.org/
  - https://packages.gentoo.org/packages/dev-cpp/robin-map
- macOS: https://formulae.brew.sh/
  - https://formulae.brew.sh/formula/robin-map#default
- Alpine: https://pkgs.alpinelinux.org/packages
  - https://alpine.pkgs.org/3.19/alpine-community-aarch64/robin-map-1.2.1-r0.apk.html
- NixOS/nixpkgs: https://search.nixos.org/packages
  - https://search.nixos.org/packages?channel=23.11&show=robin-map&from=0&size=50&sort=relevance&type=packages&query=robin
- openSUSE: https://software.opensuse.org/package/
  -  https://software.opensuse.org/package/robin-map
- rhel: https://rhel.pkgs.org/
  - https://rhel.pkgs.org/9/epel-x86_64/robin-map-devel-1.2.2-1.el9.noarch.rpm.html

### CODE REVIEW

File: rosdep/base.yaml
Comment (clalancette) [inline rosdep/base.yaml:7794]: Our typical process here is for these keys to match the name of the Ubuntu package, so this should be:

```suggestion
robin-map-dev:
```

**Review by clalancette — CHANGES_REQUESTED** [review]

File: rosdep/base.yaml
Comment (nachovizzo) [reply rosdep/base.yaml:7794]: All right. I could find many examples that contradict this, but it's not a problem at all for me so I will change it.

PS: 141 packages don't follow this convention:

ace:
acl:
antlr:
arista:
armadillo:
asio:
atlas:
binutils:
bullet-extras:
cdk:
cgal:
coinor-libcbc3:
collectd:
collectd-utils:
comedi:
cpuburn:
crypto++:
eigen2:
enblend:
enet:
f2c:
festival:
fmt:
freeimage:
geos:
glc:
glpk:
glut:
gnuplot:
gnuplot-iostream:
gperftools:
hdf5:
iwyu:
jack:
leveldb:
libapache2-mod-python:
libfftw3:
libgfortran3:
libgmp:
libgps:
libgstreamer-plugins-base0.10-0:
libgstreamer0.10-0:
libgstrtspserver-1.0-0:
libgtkmm:
libgts:
libignition-utils1:
libimage-exiftool-perl:
libjackson-json-java:
libjpeg:
libjson-glib:
libjson-java:
libjsonrpccpp-client0:
libjsonrpccpp-common0:
libjsonrpccpp-server0:
libjsonrpccpp-stub0:
libjsonrpccpp-tools:
libmagick:
libmicrohttpd:
libmockito-java:
libmongoc-1.0-0:
libnlopt0:
libnotify:
libogg:
libogre1.12.10:
libopenscenegraph:
liborocos-bfl:
libpcap:
libphonon:
libpocofoundation9:
libprocps6:
libqhull:
libqrencode:
libqt4-sql-psql:
libqwt6:
libreadline:
libstdc++6:
libsvm3:
libtesseract:
libunittest++:
libx11:
libxaw:
libxext:
libxml++-2.6:
libxml2:
libxmlrpc-c++:
libxrandr:
libxslt:
libxxf86vm:
linux-kernel-headers:
log4cplus:
lz4:
mapnik-utils:
matio:
mosquitto-clients:
mpfr:
mplayer:
mrpt:
msgpack:
multistrap:
muparser:
netcdf:
netpbm:
ninja-build:
nvidia-cuda:
nvidia-cudnn:
omniorb:
opende:
osm2pgsql:
osmium:
pcre:
pocketsphinx-bin:
poppler-utils:
portaudio:
proj:
r-base:
robin-map:
rti-connext-dds-6.0.1:
rtmidi:
sdformat13:
sdl:
sdl-gfx:
sdl-image:
sdl-mixer:
sdl-ttf:
sdl2:
sdl2-image:
sdl2-mixer:
sdl2-ttf:
setserial:
snappy:
sparsehash:
spdlog:
suitesparse:
tbb:
tinyxml2:
xdotool:
xerces:
xulrunner-1.9.2:
yaml:
zbar:
zziplib:

Comment (clalancette) [reply rosdep/base.yaml]: > All right. I could find many examples that contradict this, but it's not a problem at all for me so I will change it.

Yes, we introduced this rule a few years ago, but there are many old keys that don't follow it.  We can't really change them without breaking packages, so we tend to leave them in place.  And sometimes, we make a different choice for practical purposes.

In any case, for new keys we follow this rule where possible.

**Review by clalancette — APPROVED** [review]

**Comment by nachovizzo** [comment]
Shall we merge?

**Comment by clalancette** [comment]
> Shall we merge?

We need a second review, then we can merge.

**Review by mjcarroll — APPROVED** [review]

**Pull request merged**
